test/conformance/container_test.go

```diff
@@ -25,42 +25,150 @@ import (
 	"github.com/knative/serving/pkg/apis/serving/v1alpha1"
 	"github.com/knative/serving/test"
 	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
-// TestShouldNotHaveHooks validates that we receive an error back when attempting to create a Service that
-// specifies lifecycle hooks.
-func TestShouldNotHaveHooks(t *testing.T) {
+// TestMustNotContainerContraints tests that attempting to set unsupported fields or invalid values as
+// defined by "MUST NOT" statements from the runtime contract results in a user facing error.
+func TestMustNotContainerConstraints(t *testing.T) {
 	t.Parallel()
 	clients := setup(t)
-	names := test.ResourceNames{
-		Service: test.ObjectNameForTest(t),
-		Image:   pizzaPlanet1,
-	}
 
-	hooks := []func(s *v1alpha1.Service){
-		func(s *v1alpha1.Service) {
-			lifecycleHandler := &corev1.ExecAction{
-				Command: []string{"/bin/sh", "-c", "echo Hello from the post start handler > /usr/share/message"},
-			}
-			s.Spec.RunLatest.Configuration.RevisionTemplate.Spec.Container.Lifecycle = &corev1.Lifecycle{
-				PostStart: &corev1.Handler{Exec: lifecycleHandler},
-			}
+	testCases := []struct {
+		name    string
+		options func(s *v1alpha1.Service)
+	}{{
+		name: "TestArbitraryPortName",
+		options: func(s *v1alpha1.Service) {
+			s.Spec.RunLatest.Configuration.RevisionTemplate.Spec.Container.Ports = []corev1.ContainerPort{
+				{Name: "arbitrary",
```

Comment:
this feels live a very strange formatting.
Can you either make them one line `{a:b, c:d}` or do full indentation
```
{
  a: b,
  c: d,
}
```

Reply:
Fixed formatting